src/engine/Result.cpp

```diff
@@ -120,6 +120,9 @@ void Result::applyLimitOffset(
   // than the size of the `IdTable`, then this has no effect and runtime
   // `O(1)` (see the docs for `std::shift_left`).
   AD_CONTRACT_CHECK(limitTimeCallback);
+  if (!limitOffset._limit.has_value() && limitOffset._offset == 0) {
```

Comment:
I am not sure, but I think I once implemented something like `limitOffset.isUnconstrained()` That does exactly what you are doing manually here but is more idiomatic.